storm-client/src/jvm/org/apache/storm/utils/DefaultShellLogHandler.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.storm.utils;
+
+import org.apache.storm.multilang.ShellMsg;
+import org.apache.storm.task.TopologyContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handle output from non-JVM processes.
+ */
+public class DefaultShellLogHandler implements ShellLogHandler {
+    public static final Logger LOG = LoggerFactory.getLogger(DefaultShellLogHandler.class);
```

Comment:
Looks like we're losing class information which is logging. (I mean we no longer write log with ShellSpout/ShellBolt.) 

Given that we are creating ShellLogHandler instance per task, we can pass class name in `setUpContext()` (interface needs to be changed) and initialize logger based on the class name. Makes sense?

Reply:
Yep, makes sense - thanks!